src/praisonai-agents/tests/unit/sandbox/test_sandbox_wiring.py

```
@@ -0,0 +1,129 @@
+        """User-supplied tool with a colliding name."""
+        return "user"
+
+    agent = Agent(name="T", instructions="x", tools=[execute_python_code], sandbox=True)
+    assert _tool_names(agent).count("execute_python_code") == 1
+
+
+# ── the tools must genuinely isolate ─────────────────────────────────────────
+def test_sandboxed_code_runs_out_of_process():
+    """The point of the feature: not the same interpreter as the caller."""
+    agent = Agent(name="T", instructions="x", sandbox=True)
+    run = [t for t in agent.tools if getattr(t, "__name__", "") == "execute_python_code"][0]
```

Ruff (0.16.1)

[warning] Prefer `next(...)` over single element slice

Replace with `next(...)`

(RUF015)